registry/registry_test.go

```diff
@@ -0,0 +1,317 @@
+
+	r, err := cl.NewRegistry(ctx, req.Name, req.Type,
+		req.URL, &model.RegistryCredential{}, true)
+
+	p.AssertExpectations(t)
+	assert.Nil(t, r)
+	if assert.Error(t, err) {
+		assert.Equal(t, "illegal format of provided ID value", err.Error())
+	}
+}
+
+func TestRESTClient_GetRegistry(t *testing.T) {
+	name := "testregistry"
+	insecure := true
+	regType := "harbor"
+	url := "http://foo.bar"
+	id := int64(11)
+	ctx := context.Background()
+
+	p := &mocks.MockClientService{}
+	p.On("GetRegistries",
+		&products.GetRegistriesParams{
+			Name:    &name,
+			Context: ctx,
+		}, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc"),
+	).Return(&products.GetRegistriesOK{
+		Payload: []*model.Registry{
+			{
+				CreationTime: "",
+				Credential:   nil,
+				Description:  "",
+				ID:           id,
+				Insecure:     insecure,
+				Name:         name,
+				Status:       "",
+				Type:         regType,
+				UpdateTime:   "",
+				URL:          url,
+			},
+		},
+	}, nil)
+
+	cl := NewClient(&client.Harbor{Products: p, Transport: nil}, authInfo)
+	r, err := cl.GetRegistry(ctx, name)
+
+	p.AssertExpectations(t)
+	assert.NoError(t, err)
+	assert.Equal(t, id, r.ID)
+	assert.Equal(t, name, r.Name)
+	assert.Equal(t, url, r.URL)
+	assert.Equal(t, regType, r.Type)
+	assert.Equal(t, insecure, r.Insecure)
+}
+
+func TestRESTClient_GetRegistry_NotFound(t *testing.T) {
+	name := "testregistry"
+	ctx := context.Background()
+
+	p := &mocks.MockClientService{}
+	p.On("GetRegistries",
+		&products.GetRegistriesParams{
+			Name:    &name,
+			Context: ctx,
+		}, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc"),
+	).Return(&products.GetRegistriesOK{
+		Payload: []*model.Registry{},
+	}, nil)
+
+	cl := NewClient(&client.Harbor{Products: p, Transport: nil}, authInfo)
+	r, err := cl.GetRegistry(ctx, name)
+
+	p.AssertExpectations(t)
+	assert.Nil(t, r)
+	if assert.Error(t, err) {
+		assert.IsType(t, &ErrRegistryNotFound{}, err)
+	}
+}
+
+func TestRESTClient_GetRegistry_ErrOnGet(t *testing.T) {
+	name := "testregistry"
+	ctx := context.Background()
+	errorMsg := "error on server side"
+	errorCode := 500
+
+	p := &mocks.MockClientService{}
+	p.On("GetRegistries",
+		&products.GetRegistriesParams{
+			Name:    &name,
+			Context: ctx,
+		}, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc"),
+	).Return(nil, &runtime.APIError{
+		OperationName: "",
+		Response:      errorMsg,
+		Code:          errorCode,
+	})
+
+	cl := NewClient(&client.Harbor{Products: p, Transport: nil}, authInfo)
+	r, err := cl.GetRegistry(ctx, name)
+
+	p.AssertExpectations(t)
+	assert.Nil(t, r)
+	if assert.Error(t, err) {
+		require.IsType(t, &ErrRegistryInternalErrors{}, err)
+	}
+}
+
+func TestRESTClient_GetRegistry_ErrRegistryNotProvided(t *testing.T) {
+	p := &mocks.MockClientService{}
+	cl := NewClient(&client.Harbor{Products: p, Transport: nil}, authInfo)
+	r, err := cl.GetRegistry(context.Background(), "")
+
+	p.AssertExpectations(t)
```

Comment:
I think this can be left out, as there are no `mock.On` or `mock.Return` calls. As per the GoDoc of this method,
> AssertExpectations asserts that everything specified with On and Return was in fact called as expected.